warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1210,32 +1210,19 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             stopwatch.stop();
         }
         
-        if (!disableCompositeFields) {
-            stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand composite terms");
-            
-            try {
-                config.setCompositeToFieldMap(metadataHelper.getCompositeToFieldMap(config.getDatatypeFilter()));
-                config.setCompositeTransitionDates(metadataHelper.getCompositeTransitionDateMap(config.getDatatypeFilter()));
-                config.setCompositeFieldSeparators(metadataHelper.getCompositeFieldSeparatorMap(config.getDatatypeFilter()));
-                config.setFieldToDiscreteIndexTypes(CompositeUtils.getFieldToDiscreteIndexTypeMap(config.getQueryFieldsDatatypes()));
-            } catch (TableNotFoundException e) {
-                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_ACCESS_ERROR, e);
-                throw new DatawaveFatalQueryException(qe);
-            }
-            
-            queryTree = ExpandCompositeTerms.expandTerms(config, queryTree);
-            stopwatch.stop();
-            if (log.isDebugEnabled()) {
-                logQuery(queryTree, "Query after expanding composite terms:");
-            }
-        }
+        /**
+         * Composite expansion was here; however, we are moving this to Post bounded range/regex lookup to facilitate regexes expanding and thus being
+         * considered for composite terms.
+         */
+        queryTree = checkAndExpandCompositeTerms(config, queryTree, timers, "Pre-regex expansion");
```

Comment:
Not sure I fully follow, but is this work being done twice? Would it make sense to move to the else block for the `if (!disableBoundedLookup)` test?

Reply:
They're independent operations, actually, meaning bounded lookups could be disabled and composite expansion is enabled.

I think this is a result of how ExpandCompositeTerms is doing its work. It will place trailing regexes on the second term ( of an and ), allowing those terms to expand in the global index through the typical index lookup. In the case where we have a term like FIELD:blah* AND FIELDTWO:roses it would previously not create composite nodes. Ideally we'd expand FIELD:blah* and join the composite terms, which is why when we leave ExpandComposite terms the first time we are left with FIELD:BLAH* and FIELDTWO:ROSES. ParallelIndexExpansion expands blah* to say (FIELD:blahdy OR FIELD:blahrt ) && FIELDTWO:roses. So then the secondary composite expansion will get those EQNodes and set FIELD_FIELDTWO:blahdy\xffffroses OR FIELD_FIELDTWO:blahrt\xfffroses. 

This could be better corrected with doing the index expansion in expand composite terms AND then not doing it a second time; however, that would incur more code changes and not yield something something that is more performant.

Reply:
I think a perfectly reasonable response is : oh this is too hokey, we should clean this up, which I'm fine investigating. I created this PR to gauge desire to minimize changes and hence impact or fix it by creating a much larger PR.

Reply:
> They're independent operations, actually, meaning bounded lookups could be disabled and composite expansion is enabled.

Yeah, I got that part, but what I'm not following is if you must expand twice? What would break if you left the expansion you added on line 1244 and moved the other one from line 1217 to the else on line 1374/1372?

Reply:
Oh, and I wouldn't say it's too hokey. If it's performant and meets our use case, then I'm in favor. Getting it contributed back to the main line might take more work, though.